examples/kelvin_helmholtz_instability.jl

```diff
@@ -0,0 +1,369 @@
+
+# # Stratified Kelvin-Helmholtz instability
+#
+# We simulate Kelvin-Helmholtz instability in two-dimensions in ``x, z``,
+
+using Oceananigans
+
+grid = RegularCartesianGrid(size=(64, 1, 65), x=(-5, 5), y=(0, 1), z=(-5, 5),
+                                  topology=(Periodic, Periodic, Bounded))
+
+# # The basic state
+#
+# We're simulating the instability of a sheared and stably-stratified basic state
+# ``U(z)`` and ``B(z)``. Two parameters define our basic state: the Richardson number
+# defined as,
+#
+# ```math
+# Ri = \frac{∂_z B}{∂_z U^2}
+# ```
+#
+# and the width of the stratification layer, ``h``.
+
+using Oceananigans.Fields
+
+shear_flow(x, y, z, t) = tanh(z)
+
+stratification(x, y, z, t, p) = p.h * p.Ri * tanh(z / p.h)
+
+U = BackgroundField(shear_flow)
+B = BackgroundField(stratification, parameters=(Ri=0.1, h=1/4))
+
+# Our basic state thus has a thin layer of stratification in the center of
+# the channel, embedded within a thicker shear layer surrounded by unstratified fluid.
+
+using Plots, Oceananigans.Grids
+
+z = znodes(Cell, grid)
+
+Ri, h = B.parameters
+
+kwargs = (ylabel="z", linewidth=2, label=nothing)
+
+ U_plot = plot(shear_flow.(0, 0, z, 0), z; xlabel="U(z)", kwargs...)
+ B_plot = plot([stratification(0, 0, z, 0, (Ri=Ri, h=h)) for z in z], z; xlabel="B(z)", color=:red, kwargs...)
+Ri_plot = plot(Ri * sech.(z / h).^2 ./ sech.(z).^2, z; xlabel="Ri(z)", color=:black, kwargs...)
+
+plot(U_plot, B_plot, Ri_plot, layout=(1, 3), size=(800, 400))
+
+# # The model
+
+using Oceananigans.Advection
+
+model = IncompressibleModel(timestepper = :RungeKutta3, 
+                              advection = WENO5(),
+                                   grid = grid,
+                               coriolis = nothing,
+                      background_fields = (u=U, b=B),
+                                closure = IsotropicDiffusivity(ν=1e-4, κ=1e-4),
+                               buoyancy = BuoyancyTracer(),
+                                tracers = :b)
+
+# # A _Power_ful algorithm
+#
+# We will find the most unstable mode of the Kelvin-Helmholtz instability using the "power method". 
+# Linear instabilities, such as the Kelvin-Helmholtz instability, are described by linearized equations about a base state that take the form
+# ```math
+# \partial_t \Phi = L \Phi \, .
+# ```
+# In this example, ``\Phi = (u, v, w, b)`` is a vector of the velocities ``u, v, w``, and 
```

Comment:
I think this should be ``\Phi = (u, v, w, b, p)``, right?

Reply:
I see what you're getting at but I don't think that's the solution because `p` does not obey a prognostic equation... ? I guess either we have to write this differently (something like `A dt(Phi) = L Phi`) or explain that `Phi` is merely proportional to `u, v, w, p, b` somehow. Because I think what you've noticed is that to get the form I've written you need to project out pressure (or something).

Reply:
You are correct but let's not do `A dt(Phi) = L Phi`... no need. These few lines can't be a course on linear stability... 
Perhaps we can loosely write `\Phi = (u, v, w, b, p)` and add a citation to a textbook?